src/classes/Collection.cls

```diff
@@ -2,317 +2,494 @@ public with sharing class Collection {
 
 	private List<SObject> records;
 
+
+	// CONSTRUCT
+
 	public static Collection of(Iterable<SObject> records) {
 		return new Collection(records);
 	}
 
+
 	public static Collection of(Set<SObject> records) {
 		return new Collection(new List<SObject>(records));
 	}
 
+
 	private Collection(Iterable<SObject> records) {
 		this.records = new List<SObject>();
+
 		Iterator<SObject> iter = records.iterator();
-		while (iter.hasNext()) {
+
+		while(iter.hasNext()) {
 			this.records.add(iter.next());
 		}
 	}
 
+
+	// CONVERT
+
 	public List<SObject> asList() {
 		return new List<SObject>(records);
 	}
 
+
 	public List<SObject> asList(Type listType) {
-		List<SObject> typedList = (List<SObject>) listType.newInstance();
+		List<SObject> typedList =(List<SObject>) listType.newInstance();
 		typedList.addAll(records);
 		return typedList;
 	}
 
+
 	public Set<SObject> asSet() {
 		return new Set<SObject>(records);
 	}
 
+
 	public Set<SObject> asSet(Type setType) {
-		Set<SObject> typedSet = (Set<SObject>) setType.newInstance();
+		Set<SObject> typedSet =(Set<SObject>) setType.newInstance();
 		typedSet.addAll(records);
 		return typedSet;
 	}
 
+
+	public Map<Id, SObject> asMap() {
```

Comment:
Very happy with `asMap`, it's a previously unimplemented benefit of the rewrite done for https://github.com/ipavlic/apex-lambda/issues/4. I would just put a space between assignment and value assigned, as elsewhere (space before the cast braces).